### ISSUE TITLE
Difficult handling of join(...tasks) with dynamic number of tasks (i.e. possibly 1)

### ISSUE THREAD

**Issue by leoselig** [redux-saga/redux-saga#1554]
I'd like to discuss an API enhancement to solve an issue that we hit when working with variable sets of tasks.

## Situation
We have a set of tasks with an unknown size (>= 0).

```javascript
const severalTasks = getSeveralTasks();
const severalResults = yield join(...severalTasks);
```

## Problem

I'd expect `severalResults` to be an array. In most cases it actually is unless `severalTasks.length` is `1`. In that case, due to the use of spread args we actually end up in the `join(task)` signature and receive a single result back. This is obviously unhandy in this scenario.

## Workaround

```javascript
const severalTasks = getSeveralTasks();
const severalResults = severalTasks.length === 1 ? [yield join(...severalTasks)] : yield join(...severalTasks);
```

## Proposed Solution
By adding support for a `join(task[])` siganture the following could supported in a backwards-compatible fashion:

```javascript
const severalTasks = getSeveralTasks();
const severalResults = yield join(severalTasks);
```

From a quick look at the [API reference](https://redux-saga.js.org/docs/api/) I also feel like this is limited to `join()`.

If this is something desireable I'd be happy to contribute.

**Comment by shinima** [comment]
Actually we have discussed this problem recently in #1515, and it has been merged in #1527 . It will be available when redux-saga v1.0 gets published.

**Comment by leoselig** [comment]
Oh, I apologize then. I searched in the issues but apparently not thoroughly. Nice to know and thanks for the quick response!
